modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/RegistrationInfoImpl.java

```diff
@@ -574,15 +579,24 @@ public String getImplementation() {
         return implementation;
     }
 
-    public void checkExtensions() {
+    /**
+     * Checks extensions and returns only valid ones.
+     *
+     * @since 11.3
+     */
+    protected Extension[] checkExtensions() {
+        var validExtensions = new ArrayList<Extension>();
         for (ExtensionImpl xt : extensions) {
             if (xt.target == null) {
                 String msg = String.format(
-                        "Bad extension declaration (no target attribute specified) on component 's%'", getName());
-                Framework.getRuntime().getMessageHandler().addMessage(Level.WARNING, msg);
+                        "Bad extension declaration (no target attribute specified) on component '%s'", getName());
+                Framework.getRuntime().getMessageHandler().addMessage(Level.ERROR, msg);
                 continue;
+            } else {
+                validExtensions.add(xt);
             }
         }
+        return validExtensions.toArray(Extension[]::new);
```

Comment:
Nice, I didn't realize this had been added in Java 11, it's indeed cleaner-looking than `.toArray(new Extension[0])`